framework/wazuh/core/stats.py

```diff
@@ -2,23 +2,215 @@
+        stat_filename = os.path.join(
+            common.stats_path, "totals", str(date.year), MONTHS[date.month - 1],
+            f"ossec-totals-{date.strftime('%d')}.log")
+        stats = open(stat_filename, 'r')
+    except IOError:
+        raise WazuhError(1308, extra_message=stat_filename)
+    alerts = []
+    affected = []
+    for line in stats:
+        data = line.split('-')
+        if len(data) == 4:
+            sigid = int(data[1])
+            level = int(data[2])
+            times = int(data[3])
+            alert = {'sigid': sigid, 'level': level, 'times': times}
+            alerts.append(alert)
+        else:
+            data = line.split('--')
+            if len(data) != 5:
+                if len(data) in (0, 1):
+                    continue
+                else:
+                    return True, affected
+            hour = int(data[0])
+            total_alerts = int(data[1])
+            events = int(data[2])
+            syscheck = int(data[3])
+            firewall = int(data[4])
+            affected.append(
+                {'hour': hour,
+                 'alerts': alerts,
+                 'totalAlerts': total_alerts,
+                 'events': events,
+                 'syscheck': syscheck,
+                 'firewall': firewall
+                 })
+            alerts = []
+
+    return False, affected
+
+
+def get_daemons_stats_(filename):
+    """Get daemons stats from an input file.
+
+    Parameters
+    ----------
+    filename: str
+        Full path of the file to get information.
+
+    Returns
+    -------
+    array
+        Stats of the input file.
+
+    Raises
+    ------
+    WazuhError
+        Raised if file does not exist.
+    """
+    try:
+        with open(filename, mode='r') as f:
+            input_file = str("[root]\n" + f.read())
+        fp = StringIO(input_file)
+        config = configparser.RawConfigParser()
+        config.read_file(fp)
+        items = dict(config.items("root"))
+        try:
+            for key, value in items.items():
+                items[key] = float(value[1:-1])
+        except Exception as e:
+            return WazuhInternalError(1104, extra_message=str(e))
+    except IOError:
+        raise WazuhError(1308, extra_message=filename)
+
+    return [items]
+
+
+def get_agents_component_stats_json_(agent_list, component):
+    """Get statistics of an agent's component.
+
+    Parameters
+    ----------
+    agent_list: list, optional
+        List of agents ID's, by default None.
+    component: str, optional
+        Name of the component to get stats from, by default None.
+
+    Returns
+    -------
+    tuple
+        Failed: array of failed agent's information and related wazuh error.
+        Affected: array of agent's statistics.
+
+    Raises
+    ------
+    WazuhResourceNotFound
+        Raised if agent does not exist.
+    """
+    failed = []
+    affected = []
+    system_agents = agent.get_agents_info()
+    for agent_id in agent_list:
```

Comment:
This funcion in core should not be in charge of checking the system agents. This function (core), should just receive a single agent and a single component. The result and possible exceptions should all be checked from SDK side.